internal/resources/app_endpoint_activation_status.go

```diff
@@ -0,0 +1,368 @@
+}
+
+// retrieveAppEndpointActivation reads the app endpoint and maps its state to the Online flag.
+func (r *AppEndpointActivationStatus) retrieveAppEndpointActivation(ctx context.Context, organizationId, projectId, clusterId, appServiceId, appEndpointName string) (*app_service_api.GetAppEndpointStateResp, error) {
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s", r.HostURL, organizationId, projectId, clusterId, appServiceId, appEndpointName)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := r.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		r.Token,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var getResp app_service_api.GetAppEndpointStateResp
+	if err := json.Unmarshal(response.Body, &getResp); err != nil {
+		return nil, err
+	}
+
+	return &getResp, nil
+}
+
+// waitForAppEndpointStatus monitors the status of an App Endpoint online/offline request.
+// It periodically fetches the App Endpoint status using the `retrieveAppEndpointActivation`
+// function and waits until the App Endpoint reaches the desired state or until a specified timeout is reached.
+// The function returns an error if the operation times out or encounters an error during status retrieval.
+func (r *AppEndpointActivationStatus) waitForAppEndpointStatus(ctx context.Context, organizationId, projectId, clusterId, appServiceId, appEndpointName string, online bool) (*providerschema.AppEndpointActivationStatus, error) {
+	var (
+		appEndpointResp *app_service_api.GetAppEndpointStateResp
+		err             error
+	)
+
+	// Assuming 20 minutes is the max time online/offline takes, can change after discussion
+	const timeout = time.Minute * 20
+
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	const sleep = time.Second * 3
+	timer := time.NewTimer(10 * time.Second)
+
+	var desiredState string
+	if online {
+		desiredState = "Online"
+	} else {
+		desiredState = "Offline"
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("app endpoint activation status transition timed out after initiation, unexpected error: %w", err)
+		case <-timer.C:
+			appEndpointResp, err = r.retrieveAppEndpointActivation(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName)
+			switch err {
+			case nil:
+				if appEndpointResp.State == desiredState {
+					return &providerschema.AppEndpointActivationStatus{
+						OrganizationId:  types.StringValue(organizationId),
+						ProjectId:       types.StringValue(projectId),
+						ClusterId:       types.StringValue(clusterId),
+						AppServiceId:    types.StringValue(appServiceId),
+						AppEndpointName: types.StringValue(appEndpointName),
+						Online:          types.BoolValue(online),
+					}, nil
+				}
+			default:
+				return nil, err
+			}
+			timer.Reset(sleep)
```

Comment:
nit

can simplify to something like this (assuming func changed to just return an err) :

```
appEndpointResp, err = r.retrieveAppEndpointActivation(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName)

if err != nil { return err }

if appEndpointResp.State == desiredState { return nil }

timer.Reset(sleep)